eng/pipelines/templates/jobs/archetype-sdk-client.yml

```
@@ -79,18 +88,37 @@ jobs:
   - checkout: self
     submodules: recursive
 
+  # Install apt dependencies (if appropriate)
+  - bash: sudo apt install -y $(AptDependencies)
+    condition: ne(variables['AptDependencies'], '')
+    displayName: Install dependencies from apt
+
   - template: /eng/pipelines/templates/steps/vcpkg.yml
     parameters:
       DependenciesVariableName: VcpkgInstall
 
+  - script: |
+      dotnet tool install -g dotnet-reportgenerator-globaltool
+      dotnet tool install dotnet-reportgenerator-globaltool --tool-path tools
+    displayName: Install coverage tools
+    condition: ne(variables['CODE_COVERAGE'], '')
```

```suggestion
    condition: and(succeededOrFailed(), ne(variables['CODE_COVERAGE'], ''))
```

Need succeededOrFailed as part of the conditions otherwise these steps will still try to execute in the case of a canceled build. 

I think this is probably the condition you want in your code-coverage template.